oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/parser/ExceptionAnalyzerStrategy.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.analyzer.provider.trace.parser;
+
+import lombok.Getter;
+import org.apache.skywalking.oap.server.analyzer.provider.trace.parser.listener.strategy.AnyExceptionAnalyzer;
+import org.apache.skywalking.oap.server.analyzer.provider.trace.parser.listener.strategy.EntryFirstExceptionAnalyzer;
+import org.apache.skywalking.oap.server.analyzer.provider.trace.parser.listener.strategy.ExceptionAnalyzer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * define the strategy to analysis segment status
+ */
+public enum ExceptionAnalyzerStrategy {
+    ANY(AnyExceptionAnalyzer.class),
+
+    ENTRY_FIRST(EntryFirstExceptionAnalyzer.class);
+
+    private final Logger log = LoggerFactory.getLogger(ExceptionAnalyzerStrategy.class);
+    @Getter
+    private ExceptionAnalyzer exceptionAnalyzer;
+
+    ExceptionAnalyzerStrategy(final Class<? extends ExceptionAnalyzer> clazz) {
+        try {
+            this.exceptionAnalyzer = clazz.newInstance();
```

Comment:
Even reflect cost is limited, but it is not necessary. `ExceptionAnalyzer` implementors are stateless, you could have the instance in the constructor directly.